starport/pkg/cosmosgen/templates/module/module.ts.tpl

```diff
@@ -0,0 +1,61 @@
+// THIS FILE IS GENERATED AUTOMATICALLY. DO NOT MODIFY.
+
+import { StdFee } from "@cosmjs/launchpad";
+import { SigningStargateClient, DeliverTxResponse } from "@cosmjs/stargate";
+import { EncodeObject } from "@cosmjs/proto-signing";
+
+import { Api } from "./rest";
+{{ range .Module.Msgs }}import { {{ .Name }} } from "./types/{{ resolveFile .FilePath }}";
+{{ end }}
+
+{{ range .Module.Msgs }}
+type send{{ .Name }}Params = {
+  value: {{ .Name }},
+  fee?: StdFee,
+  memo?: string
+};
+{{ end }}
+{{ range .Module.Msgs }}
+type {{ camelCase .Name }}Params = {
+  value: {{ .Name }},
+};
+{{ end }}
+
+class Module extends Api<any> {
+	private _client: SigningStargateClient;
+	private _address: string;
+
+  	constructor(baseUrl: string) {
+		super({
+			baseUrl
+		})
+	}
+
+	public withSigner(client: SigningStargateClient, address: string) {
+		this._client = client;
+		this._address = address;
+	}
+
+	{{ range .Module.Msgs }}
+	async send{{ .Name }}({ value, fee, memo }: send{{ .Name }}Params): Promise<DeliverTxResponse> {
```

Comment:
I was checking this func, seems like there is a sendX func for each SDK message and inside implementation we call signAndBroadcast?

In the previous design of plain TS client, msgX functions was constructing the SDK message and user needs to call signAndBroadcast. This also makes it possible user to send multiple messages at once inside the single TX.

I think we should keep the all old API/behaviour for two reasons:
* We need to support broadcasting TXs with more than one message inside.
* Above, also makes it possible to revert changes if one of the messages fail inside the TX.
* It's a lot faster.

I'm just reviewing through the template so I might understood it wrong so please correct if the API already supports what I mentioned above.